iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -284,4 +334,42 @@ private static String getCatalogType(Configuration conf, String catalogName) {
       }
     }
   }
+
+  public static class TableCache {
+    private static final Cache<String, Table> tableCache = Caffeine.newBuilder()
+        .expireAfterAccess(12, TimeUnit.HOURS).build();
+    private static final Logger LOG = LoggerFactory.getLogger(TableCache.class);
+
+    public static void removeTables(Configuration conf) {
+      String queryId = conf.get(HiveConf.ConfVars.HIVEQUERYID.varname);
+      if (queryId != null && !queryId.isEmpty()) {
+        Set<String> queryKeys = tableCache.asMap().keySet().stream()
+            .filter(k -> k.startsWith(queryId)).collect(Collectors.toSet());
+        tableCache.invalidateAll(queryKeys);
+      } else {
+        LOG.warn("Query id is not present in config, therefore no Iceberg table object is removed " +
```

Comment:
Is this really a `warn`? Do we expect at least one table it the cache every time?

Reply:
No, we don't except, but I guess we should signal that the query id is not present config, which could lead to cache pollution.

Reply:
I think what we do expect is that the `hive.query.id` is always populated in the config. If it's absent, that's a warning sign and something which could prevent us from evicting from the cache. This could be a useful log line in case we see that the cache gets overly large and we're looking for the root cause why the iceberg tables are not evicted.